wisp-logging-testing/src/main/kotlin/wisp/logging/LogCollector.kt

```diff
@@ -43,7 +43,17 @@ interface LogCollector {
   fun takeMessages(
     loggerClass: KClass<*>? = null,
     minLevel: Level = Level.INFO,
-    pattern: Regex? = null
+    pattern: Regex? = null,
+  ): List<String>
+
+  /**
+   * Takes all matching messages, optionally leaving unmatched logs in this collector.
+   */
+  fun takeMessages(
+    loggerClass: KClass<*>? = null,
+    minLevel: Level = Level.INFO,
+    pattern: Regex? = null,
+    consumeUnmatchedLogs: Boolean = false,
   ): List<String>
```

Comment:
I think maybe this is the interface we want to land on eventually if some folks depend on consuming all unmatched logs?

Reply:
Yep, looks good.